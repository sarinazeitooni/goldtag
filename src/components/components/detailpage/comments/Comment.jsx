import React from 'react';
import {MobileView} from "react-device-detect";
import LikeAndDislike from "./likes/LikeAndDislike";
import commentTexts from "./texts/commentTexts";
import Divider from "../../divider/Divider";
function Comment({name, date, customercomment, goldcomment,like,dislike}){
    return(
        <div className='QA-container'>
            <div className="customer">
                <div className="customer-info">
                    <img src="https://cdn.goldtag.net/web/uploads/pictures/users/thumb/avatar_women.jpg" alt="customer"/>
                    <div className="customer-text-info">
                        <h3 className="customer-name">{name}</h3>
                        <div className="date">{date}</div>
                    </div>
                </div>
                <p className="customer-comment">{customercomment}</p>
                <LikeAndDislike like={like} dislike={dislike}/>
            </div>
            <div className='gold'>
                <MobileView className='gold-info'>
                    <div className="gold-info-text">
                        <h3 className="customer-name-gold">{commentTexts.goldTag}</h3>
                        <div className="date">{date}</div>
                    </div>
                </MobileView>
                <img className='gold-avatar' src='https://cdn.goldtag.net/web/uploads/pictures/users/thumb/1_1556417043299.jpg' alt="gold"/>
                <p className="goldcomment">{goldcomment}</p>
            </div>
            <Divider className='dashed'/>
        </div>
    )
};
export default Comment;