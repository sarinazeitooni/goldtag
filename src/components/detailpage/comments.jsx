import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {faComment,faComments} from "@fortawesome/free-solid-svg-icons";
import '../../style/gold-comments.scss';
import '../../style/customer-comments.scss';
import '../../style/comments-container.scss';
import {BrowserView , MobileView} from 'react-device-detect';
function Comments({id}){
    const data = id.comments;
    return(
        <div className='comments-container'>
            <a className='QandA'><FontAwesomeIcon icon={faComments}></FontAwesomeIcon> پرسش و پاسخ </a>
            <a><FontAwesomeIcon icon={faComment}></FontAwesomeIcon> نظرات مشتریان </a>
            <hr ></hr>
            <div className='QandA-container'>
                    {data.map((item)=>{
                        return(
                            <div className='QA-container'>
                                <div className="customer">
                                    <div className="customer-info">
                                        <img src="https://cdn.goldtag.net/web/uploads/pictures/users/thumb/avatar_women.jpg" alt="customer"/>
                                        <div className="customer-text-info">
                                            <h3 className="customer-name">{item.name}</h3>
                                            <div className="date">{item.date}</div>
                                        </div>
                                    </div>
                                    <p className="customer-comment">{item.customercomment}</p>
                                </div>

                                <div className='gold'>

                                    <MobileView className='gold-info'>
                                        <div className="gold-info-text">
                                        <h3 className="customer-name-gold">گلدتگ</h3>
                                            <div className="date">{item.date}</div>
                                        </div>
                                    </MobileView>
                                    <img className='gold-avatar' src='https://cdn.goldtag.net/web/uploads/pictures/users/thumb/1_1556417043299.jpg' alt="gold"/>
                                    <p className="goldcomment">{item.goldcomment}</p>
                                </div>
                                <hr className='dashed'></hr>
                            </div>
                        )
                    })}
            </div>
            <div className="customer-opinion-container">
                <div className='rate-container'>
                    <h3>ارسال نظر!</h3>
                    <p>
                    برای ارسال نظر باید وارد سایت شوید. برای این کار از گزینه ورود در منو بالا استفاده نمایید.
                    </p>
                </div>
            </div>
        </div>
    );
}
export default Comments