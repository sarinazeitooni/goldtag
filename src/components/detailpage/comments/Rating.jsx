import commentTexts from "./texts/commentTexts";
import React from "react";
function Rating(){
    return(
        <div className="customer-opinion-container">
            <div className='rate-container'>
                <h3>{commentTexts.rate}</h3>
                <p>{commentTexts.rateRule}</p>
            </div>
        </div>
    )
}
export default Rating;