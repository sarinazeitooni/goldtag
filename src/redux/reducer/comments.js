const initialState = {
    Tab : false
}
export default function comments(state=initialState, action) {
    switch (action.type) {
        case 'Comments':
            return {
                Tab : true,
            };
        case 'Rating':
            return{
                Tab : false
            }
        default:
            return state;
    }
}