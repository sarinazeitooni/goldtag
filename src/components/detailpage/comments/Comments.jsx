import React from 'react';
import './style/gold-comments.scss';
import './style/customer-comments.scss';
import './style/comments-container.scss';
import Comment from "./Comment";
function Comments({items}){
    const data = items.comments;
    return(
            <div className='QandA-container'>
                    {data.map((item)=>{
                        return(
                            <div key={item.id}>
                                <Comment id={item.id} like={item.like} dislike={item.dislike} name={item.name} date={item.date} customercomment ={item.customercomment} goldcomment={item.goldcomment} />
                            </div>
                        )
                    })}
            </div>
    );

}
export default Comments