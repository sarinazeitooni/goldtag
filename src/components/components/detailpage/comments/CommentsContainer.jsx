import React, {useEffect, useState} from 'react';
import {useDispatch,useSelector } from "react-redux";
import Comments from "./Comments";
import Rating from "./Rating";
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import {faComment, faComments} from "@fortawesome/free-solid-svg-icons";
import commentTexts from "./texts/commentTexts";
import actionTypes from "../../../redux/actionTypes/actionTypes";
function CommentsContainer({id}){
    const dispatch = useDispatch();
    function Dispatch(action){
        dispatch({
            type : action
        })
    }
    let target;
    function style(e){
        target = e.target;
        target.classList.add('clicked');
    }
    function showComments(e){
        Dispatch(actionTypes.Comments);
        style(e);
    };
    function showRating(e){
        Dispatch(actionTypes.Rating);
        style(e);
    };
    const Tab = useSelector(state => state.comments.Tab);
    localStorage.setItem('comments' , JSON.stringify(id.comments));
    return(
        <div className='comments-container'>
                <a className={ (Tab) ? "clicked Tabs" : 'unClicked Tabs'} onClick={showComments}>
                    <FontAwesomeIcon icon={faComments}></FontAwesomeIcon>{commentTexts.QandA}
                </a>
                <a className={ (Tab) ? 'unClicked Tabs' : "clicked Tabs"} onClick={showRating}>
                    <FontAwesomeIcon icon={faComment}></FontAwesomeIcon>{commentTexts.comments}
                </a>
                <hr></hr>
            {(!Tab) ? <Comments items={id}/> : ''}
            <Rating/>
        </div>
    )
}
export default  CommentsContainer;