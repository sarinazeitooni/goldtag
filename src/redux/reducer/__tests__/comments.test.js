import React from 'react';
import {shallow} from 'enzyme';
import comments from "../comments";
describe("comments reducer component", () => {

    const container = shallow(<comments/>)
    it("it should render", () => {
        expect(container).toMatchSnapshot();
    });
});