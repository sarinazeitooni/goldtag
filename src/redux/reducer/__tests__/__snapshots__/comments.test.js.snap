// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comments reducer component it should render 1`] = `ShallowWrapper {}`;
